File: backend/src/main/java/com/al7afez/al7afez/model/entities/MistakeType.java
package com.al7afez.al7afez.model.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;

@Entity
public class MistakeType extends MasterFile {
    @ManyToOne
    @JoinColumn(name = "parent_id")
    private MistakeType parent;

    public MistakeType getParent() {
        return parent;
    }

    public void setParent(MistakeType parent) {
        this.parent = parent;
    }
}

File: flutter_app/lib/models/mistake_type.dart
class MistakeType {
  final int id;
  final String name;
  final int? parentId;
  final String? parentName;

  const MistakeType({
    required this.id,
    required this.name,
    this.parentId,
    this.parentName,
  });

  factory MistakeType.fromJson(Map<String, dynamic> json) {
    final parent = json['parent'] as Map<String, dynamic>?;
    return MistakeType(
      id: json['id'] as int,
      name: (json['name'] as String?) ?? '',
      parentId: parent?['id'] as int?,
      parentName: parent?['name'] as String?,
    );
  }
}
